### PR TITLE
Fixed the method addAfterNode to handle the case where elementIntheLi…

### LINKED ISSUE #44: CLL - AddAfterNode - Edge case
@ani03sha In addAfterNode() method for CLL [https://github.com/ani03sha/Algorithms/blob/23352a535037446b0baaaa525d11bbf3b0fdbe15/src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java#L97](url) , shouldn't there be an additional case to check if the elementInTheList is a tail, so that the tail pointer can also be updated to the new node?

In the current implementation a test case like below will fail for this method, since it will still report 10 as the tail instead of 100: 
       ` for (int i = 1; i <= 10; i++) {
            testCircularLinkedList.add(i);
        }
        testCircularLinkedList.addAfterNode(100,10);
        assertEquals(100,testCircularLinkedList.tail.data);`

### PR DESCRIPTION
**Pull request by vaibhav26sharma**
Fix for #44 

Fixed the method addAfterNode to handle the case where elementIntheList is the tail.

Also updated the testcase covering that edge case.

### CODE REVIEW

File: src/test/java/org/redquark/algorithms/datastructures/lists/CircularLinkedListTest.java [hunk 1]
```diff
@@ -72,6 +72,8 @@ public void testAddAfterNode() {
         assertEquals(11, testCircularLinkedList.head.next.next.next.next.next.data);
         assertEquals(6, testCircularLinkedList.head.next.next.next.next.next.next.data);
         assertThrows(IllegalArgumentException.class, () -> testCircularLinkedList.addAfterNode(20, 12));
+        testCircularLinkedList.addAfterNode(100,10);
```
Comment (ani03sha) [inline src/test/java/org/redquark/algorithms/datastructures/lists/CircularLinkedListTest.java:75]: Can we add space between *100,* and *10*.

File: src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java [hunk 2]
```diff
@@ -95,6 +95,10 @@ public void addAfterNode(T element, T elementInTheList) {
         }
         // Create a new node with the given data
         ListNode<T> newNode = new ListNode<>(element);
+        if(temp == tail) {
```
Comment (ani03sha) [inline src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java:98]: Can we add proper spacing between `if` and `(`

File: src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java [hunk 3]
```diff
@@ -95,6 +95,10 @@ public void addAfterNode(T element, T elementInTheList) {
...
+            tail = newNode;
+            tail.next = head;
+        }
```
Comment (ani03sha) [inline src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java:101]: Should we return it from here?

File: src/test/java/org/redquark/algorithms/datastructures/lists/CircularLinkedListTest.java (hunk 1 above)
Comment (vaibhav26sharma) [reply src/test/java/org/redquark/algorithms/datastructures/lists/CircularLinkedListTest.java:75]: Updated the PR!

File: src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java (hunk 2 above)
Comment (vaibhav26sharma) [reply src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java:98]: Updated the PR!

File: src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java (hunk 3 above)
Comment (vaibhav26sharma) [reply src/main/java/org/redquark/algorithms/datastructures/lists/CircularLinkedList.java:101]: Updated the PR!
